acme/acme/crypto_util_test.py

```diff
@@ -10,11 +10,16 @@
 
 import josepy as jose
 import OpenSSL
+import pytest
 
 from acme import errors
 from acme import test_util
 from acme.magic_typing import List # pylint: disable=unused-import, no-name-in-module
 
+# turns all ResourceWarnings into errors for this module
+if six.PY3:
+    pytestmark = pytest.mark.filterwarnings("ignore::ResourceWarning") # pragma: no cover
```

Comment:
Almost all of the legitimate ResourceWarnings I cleaned up were in tests, with the exception of a couple one line `open().read()` situations in certbot. If we were diligent about that when reviewing code, I would be comfortable turning off the warning across the system. These acme tests are flaky about this, I think it might be a garbage collection thing but I didn't look too closely because they're inconsistent and also tests.

But, bad news, `ResourceWarning` only exists in Python 3, not 2. So we can't mention it in `pytest.ini`, because that's used across tests. If we wanted to ignore it across the board, we'd have to stick the ignore in explicitly for only python3 tests, which doesn't seem great.

Another thing fixed most easily by dropping 2.7 support!